packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/AppConfiguration.kt

```diff
@@ -268,6 +286,32 @@ public interface AppConfiguration {
             this.httpLogObfuscator = httpLogObfuscator
         }
 
+        /**
+         * Sets the name of the HTTP header used to send authorization data in when making requests to
+         * Atlas App Services. The Atlas App or firewall must have been configured to expect a
+         * custom authorization header.
+         *
+         * The default authorization header is named [DEFAULT_AUTHORIZATION_HEADER_NAME].
+         *
+         * @param name name of the header.
+         * @throws IllegalArgumentException if an empty [name] is provided.
+         */
+        public fun authorizationHeaderName(name: String): Builder = apply {
+            require(name.isNotEmpty()) { "Non-empty 'name' required." }
+            authorizationHeaderName = name
+        }
+
+        /**
+         * Update the custom headers that would be appended to every request to an Atlas App Services Application.
+         *
+         * @param block lambda with the the custom header map update instructions.
+         */
+        public fun customRequestHeaders(
+            block: MutableMap<String, String>.() -> Unit,
+        ): Builder = apply {
+            customRequestHeaders.block()
```

Comment:
Should we also check if the header name is valid or non-empty? At least we threw IllegalArgumentException in the Java implementation for that input.

Reply:
yes, we should check for empty contents.